src/wp-admin/edit-form-blocks.php

```diff
@@ -229,6 +229,10 @@ static function( $classes ) {
 	}
 }
 
+if ( wp_is_block_theme() ) {
```

Comment:
Does it need to be guarded? How does it relate to other settings like:

https://github.com/WordPress/wordpress-develop/blob/07b8ba665a64bc92ded0a3096aa8298ee2829b3a/src/wp-admin/edit-form-blocks.php#L201

I don't know myself, it might be good as is.

Reply:
I ported this check from the plugin, so not 100% sure as well - https://github.com/WordPress/gutenberg/blob/b3837134a6ecea321b70362b1ce2ab27ef601e96/lib/editor-settings.php#L27-L29

Reply:
Applying `remove_theme_support( 'block-templates' );` to Twenty Twenty-Two disables post formats per:
https://github.com/WordPress/gutenberg/blob/b3837134a6ecea321b70362b1ce2ab27ef601e96/lib/editor-settings.php#L185

However, without a guard:

```php
if ( wp_is_block_theme() ) {
```
- A `defaultTemplatePartAreas` key is still output in the `wp-edit-post-js-after` inline script.

With a guard: 

```php
if ( wp_is_block_theme() && $editor_settings['supportsTemplateMode'] ) {
```
- A `defaultTemplatePartAreas` key is no longer output in the `wp-edit-post-js-after` inline script.

Regarding the use of `$editor_settings['supportsTemplateMode']` instead of `current_theme_supports( 'block-templates' )`:
1. `$editor_settings['supportsTemplateMode']` is set via a call to `current_theme_supports( 'block-templates' )`, so the results are the same.
2. By not calling `current_theme_supports()` again, this is more performant.
3. `$editor_settings` keeps things in context.
4. `supportsTemplateMode` is clear to understand.

Happy to get feedback on that though.

While the additional key doesn't appear to break anything, it does seem to be superfluous output, so I think the guard would be a good improvement here 👍

Reply:
Thanks for the feedback. I addressed it in 3de7d48b51ff0feba9b291d39f44188ec9472cf6.

@gziolo, I think we need to backport improved check back to the plugin as well.

Reply:
Yes, it makes sense 👍🏻